.github/workflows/binaries.yml

```diff
@@ -0,0 +1,66 @@
+name: Binaries Build
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-binaries:
+    name: Build Binaries
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~1.19'
+
+      - name: Build All Binaries
+        run: make build
+
+      - name: Archive Binaries
+        uses: actions/upload-artifact@v2
+        with:
+          name: binaries
+          path: bin/
+          retention-days: 1
+
+  test:
+    needs: build-binaries
+    name: Check ${{ matrix.binary }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            binary: fioctl-linux-amd64
+          - os: ubuntu-latest
+            binary: fioctl-linux-arm64
+            qemu: true
+            qemu-platform: aarch64
+          - os: windows-latest
+            binary: fioctl-windows-amd64.exe
+          - os: macos-latest
+            binary: fioctl-darwin-amd64
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Set up QEMU
+        if: matrix.qemu
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: ${{ matrix.qemu-platform }}
+
+      - name: Download Binaries
+        uses: actions/download-artifact@v2
+        with:
+          name: binaries
+          path: bin/
+
+      - name: Set Execute Permission
+        if: runner.os != 'Windows'
+        run: chmod +x ./bin/${{ matrix.binary }}
+
+      - name: Verify binary
+        run: ./bin/${{ matrix.binary }} help
```

Comment:
I planned to write a comment that QEMU part of the script was lost.

But, it somehow magically works without QEMU, right?
FYI in my local env the `fioctl-linux-arm64 help fails to load`, so I'm puzzled how this could work at all.

Is there any more magic to the `setup-qemu-action` in play here?

Reply:
Hi @vkhoroz,

You raise a good point 🥇 . 
I removed the QEMU setup and did a test, and I observed that the arm64 failed:
[View the action run](https://github.com/camilamacedo86/fioctl/actions/runs/6386727467/job/17334072060)

![Screenshot 2023-10-03 at 01 00 04](https://github.com/foundriesio/fioctl/assets/7708031/74adfcbb-6d39-4606-ad84-66f18a9e116d)
![Screenshot 2023-10-03 at 00 59 52](https://github.com/foundriesio/fioctl/assets/7708031/b6f66b63-0f4e-44d0-818b-8f32c468f4e5)

Therefore, it seems that keeping only the following works **but it does not make sense for me either**:

```yaml
      - name: Set up QEMU
        if: matrix.qemu
        uses: docker/setup-qemu-action@v3
        with:
          platforms: ${{ matrix.qemu-platform }}
```

However, in my opinion, it is not transparent and reliable. Hence, I believe we should use a static setup so that we are making it very clear how things are working.

```yaml
      - name: Install QEMU user static
        if: matrix.qemu
        run: sudo apt-get update && sudo apt-get install -y qemu-user-static
        shell: bash

      - name: Verify binary
        run: |
          if [[ "${{ matrix.qemu }}" == "true" ]]; then
            qemu-${{ matrix.qemu-platform }}-static ./bin/${{ matrix.binary }} help
          else
            ./bin/${{ matrix.binary }} help
          fi
        shell: bash
```

Let me know your thoughts.
Can we move forward with this one now or would you like to propose any further change?

Reply:
I like how the PR looks like now.
Good to know that action _really does the magic_; it makes me feel relieved.

IMHO using apt install directly is way more unreliable, at least it is not cross-platform.

So, let's merge the shorter version (before your latest commit below).

Reply:
Done as suggested.